controllers/spacerequest/spacerequest_controller.go

```diff
@@ -43,12 +61,393 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, memberClusters map[strin
+	// it means that we couldn't find the spacerequest object on any of the given member clusters,
+	// let's return the error.
+	if err != nil && errors.IsNotFound(err) {
+		return reconcile.Result{}, errs.Wrap(err, "unable to find SpaceRequest")
+	}
+
+	if util.IsBeingDeleted(spaceRequest) {
+		logger.Info("spaceRequest is being deleted")
+		return reconcile.Result{}, r.ensureSpaceRequestDeletion(logger, memberClusterWithSpaceRequest, spaceRequest)
+	}
+	// Add the finalizer if it is not present
+	if err := r.addFinalizer(logger, memberClusterWithSpaceRequest, spaceRequest); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	subSpace, err := r.ensureSpace(logger, memberClusterWithSpaceRequest, spaceRequest)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// update spaceRequest conditions and target cluster url
+	err = updateSpaceRequest(memberClusterWithSpaceRequest, spaceRequest, subSpace)
+
+	return ctrl.Result{}, err
+}
+
+// updateSpaceRequest updates conditions and target cluster url of the spaceRequest with the ones on the Space resource
+func updateSpaceRequest(memberClusterWithSpaceRequest cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest, subSpace *toolchainv1alpha1.Space) error {
+	// reflect space conditions and target URL to spaceRequest conditions
+	spaceRequest.Status.Conditions = subSpace.Status.Conditions
+	spaceRequest.Status.TargetClusterURL = subSpace.Status.TargetCluster
+	return memberClusterWithSpaceRequest.Client.Status().Update(context.TODO(), spaceRequest)
+}
+
+// setFinalizers sets the finalizers for Space
+func (r *Reconciler) addFinalizer(logger logr.Logger, memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest) error {
+	// Add the finalizer if it is not present
+	if !util.HasFinalizer(spaceRequest, toolchainv1alpha1.FinalizerName) {
+		logger.Info("adding finalizer on SpaceRequest")
+		util.AddFinalizer(spaceRequest, toolchainv1alpha1.FinalizerName)
+		if err := memberCluster.Client.Update(context.TODO(), spaceRequest); err != nil {
+			return errs.Wrap(err, "error while adding finalizer")
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) ensureSpace(logger logr.Logger, memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest) (*toolchainv1alpha1.Space, error) {
+	logger.Info("ensuring Space", "SpaceRequest", spaceRequest.Name)
+
+	// validate tierName
+	if err := r.validateNSTemplateTier(logger, spaceRequest.Spec.TierName); err != nil {
+		return nil, err
+	}
+
+	// find parent space from namespace labels
+	parentSpace, err := r.getParentSpace(logger, memberCluster, spaceRequest)
+	if err != nil {
+		logger.Error(err, "unable to get parentSpaceName")
+		return nil, err
+	}
+	// parentSpace is being deleted
+	if util.IsBeingDeleted(parentSpace) {
+		return nil, errs.New("parentSpace is being deleted")
+	}
+
+	// get subSpace resource
+	spaceList, err := r.listSubSpaces(spaceRequest)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I've moved it at the top of the `ensureSpace` function in [df7bfda](https://github.com/codeready-toolchain/host-operator/pull/770/commits/df7bfda7359f77a69b1b9522efa41c16b7b1eb30)

It was not the first check because I was thinking that the `spacerequest.Spec.TierName` being empty may be an easier issue to validate which doesn't involve any call to the api server. 

Please let me know if this addresses your suggestion since I'm not totally sure I understood what object are you referring to when saying: 
> ... only when it doesn't exist,